src/main.cpp

```diff
@@ -173,8 +173,21 @@ int main(int argc, char **argv) {
     init_ui();
   }
 
-  if (NEEDROOT && (geteuid() != 0))
+#if NEEDROOT == 1
+  if (geteuid() != 0)
     forceExit(false, "You need to be root to run NetHogs!");
+#else
+  char exe_path[PATH_MAX];
+  unsigned int caps[5] = {0};
+
+  readlink("/proc/self/exe", exe_path, PATH_MAX);
```

Comment:
I don't know mac os well enough to know if they have file attributes like this, I think it doesn't have that feature though.